File: Data Structures & Algorithms/merge-k-sorted-linked-lists/submission-2.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next

class Solution:    
    def mergeKLists(self, lists: List[Optional[ListNode]]) -> Optional[ListNode]:
        min_index=-1
        min_num=10000
        for i in range(len(lists)):
            cur_list=lists[i]
            if cur_list and cur_list.val<min_num:
                min_index=i
                min_num=cur_list.val
            
        if min_num==10000:
            return None
        newHead=lists[min_index]
        lists[min_index]=lists[min_index].next
        cur_node=newHead

        while True:
            min_index=-1
            min_num=10000
            for i in range(len(lists)):
                cur_list=lists[i]
                if cur_list and cur_list.val<min_num:
                    min_index=i
                    min_num=cur_list.val
            
            if min_num==10000:
                return newHead
            cur_node.next=lists[min_index]
            cur_node=cur_node.next
            lists[min_index]=lists[min_index].next

File: Data Structures & Algorithms/remove-node-from-end-of-linked-list/submission-2.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */

class Solution {
public:
    ListNode* removeNthFromEnd(ListNode* head, int n) {
       ListNode* curNode=head;
       int size=1;
       while(curNode->next!=nullptr) {
        curNode=curNode->next;
        size++;
       } 
        
        //cout<<size<<endl;
       if(size==1 && n==1)return nullptr;
       int num=size-n;
       if(num==0)return head->next;

       //int pos=num-1;
       int s=1;
       curNode=head;
       while(s!=num) {
            curNode=curNode->next;
            s++;
       }

       curNode->next=curNode->next->next;
       return head;
    }
};
